### PR TITLE
Fixes 404 in add Company House company flow

### PR DESCRIPTION
**Pull request by jim68000**
This is just to unblock the ability to create companies. It reveals two further issues which will be dealt with in future commits: no labels on the company detail panel, and no error messages when an invalid DH company is sent to the back end. But I have other blockers to clear first.

### CODE REVIEW

File: src/apps/companies/controllers/ltd.js [hunk 1]
```diff
@@ -54,8 +54,9 @@ function editCommon (req, res, next) {
 }
 
 async function addDetails (req, res, next) {
+  const companyNumber = req.params.company_number || req.params.id
```
Comment (feedmypixel) [inline src/apps/companies/controllers/ltd.js]: is `req.params.company_number` still a thing here? its just `req.params.id` right?

File: src/apps/companies/router.js [hunk 2]
```diff
@@ -42,6 +42,11 @@ router
   .post(foreignController.postDetails)
 
 router
+  .route('/add/ltd/:id')
```
Comment (feedmypixel) [inline src/apps/companies/router.js]: using `.route('/add/ltd/:company_number')` could give a little more clarity?

File: src/apps/companies/controllers/ltd.js (hunk 1 above)
Comment (jim68000) [inline src/apps/companies/controllers/ltd.js]: There's another route that uses this without the :id path param, but I don't know if it actually gets called, so maybe I'm being  over cautious. Will investigate.

File: src/apps/companies/router.js (hunk 2 above)
Comment (jim68000) [inline src/apps/companies/router.js]: Agreed, will change

File: src/apps/companies/controllers/ltd.js (hunk 1 above)
Comment (jim68000) [inline src/apps/companies/controllers/ltd.js]: Actually, change below fixes this...on it

**Comment by jim68000** [comment]
Mucked up. Sorry.

**Pull request closed without merging**